plutus-core/plutus-core/examples/PlutusCore/Examples/Builtins.hs

```diff
@@ -182,18 +182,18 @@ instance uni ~ DefaultUni => ToBuiltinMeaning uni ExtensionFun where
 
     toBuiltinMeaning IdFInteger =
         makeBuiltinMeaning
-            (Prelude.id :: fi ~ Opaque term (TyAppRep f Integer) => fi -> fi)
+            (Prelude.id :: fi ~ Opaque val (TyAppRep f Integer) => fi -> fi)
```

Comment:
This is purely cosmetic though, right?  We _could_ pass terms (or anything else) to buitlins if we wanted to make PLC lazy or something, but in the current setting the things we're passing are always "values" of some sort. This is the thing I was struggling to find a word for in the specification.

Reply:
The change does make things clearer though.

Reply:
>  We could pass terms (or anything else) to buitlins if we wanted to make PLC lazy or something, but in the current setting the things we're passing are always "values" of some sort.

Not quite... So a couple years ago this whole machinery was expecting an evaluator to be passed along with a term to unlift from. The evaluator would turn the term into a value and then the user could perform the unlifting. Now that we don't pass any evaluators, there's no way we could turn a term into a value. Meaning if you try to unlift a `1 + 1`, you don't know how to evaluate it to `2` to unlift the constant. I.e. if `1 + 1` is attempted to be unlifted, there's no way you could do that within the current machinery, irrespective of whether the language is strict or lazy. So only values can be passed to the unlifting machinery.

But it's fine to lift a constant into a term rather than a value. It's also fine to unlift from a `Term` as long as that term is in fact a value represented as a `Term`.

I.e. it's fine to implement `KnownTypeIn uni val` instances for `val`s that are general terms rather than values, but then it's your responsibility to ensure that only values represented as terms are passed into the instance, because there's no way the instance could handle a non-value.